### ISSUE TITLE
DataStore keeps searching for non-existent Subscriptions

### ISSUE THREAD

**Issue by bastiancleverfit** [aws-amplify/amplify-js#5648]
**Describe the bug**
I have a **@model** called _Tenant_ where I have disabled the onCreate Subscription, after pushing to AppSync I generated the DataStore models again but it fails on the frontend giving me:

> DataStore - Sync error subscription failed Connection failed: {"errors":[{"message":"Validation error of type FieldUndefined: Field 'onCreateTenant' in type 'Subscription' is undefined @ 'onCreateTenant'"}]}

**To Reproduce**
Simply disable the subscription inside the **@model** directive and check the frontend for errors.

**Expected behavior**
I expect DataStore to not use the Subscription again.

**Code Snippet**
I defined only the Subscriptions that I need for the model (GraphQL schema):
```
type Tenant
@model(subscriptions: {
  onUpdate: ["onUpdateTenant"],
  onDelete: ["onDeleteTenant"]
})
...
```

**Smartphone (please complete the following information):**

- Device: MacBook Pro
- OS: macOS Catalina 10.15.4
- Browser: Google Chrome
- Version: 81.0.4044.129

**Comment by manueliglesias** [comment]
Hi @bastiancleverfit 

What's the scenario that you're trying to solve for, and how does disabling `onCreate` solve it for you?

**Comment by vrebo** [comment]
> Hi @bastiancleverfit
> 
> What's the scenario that you're trying to solve for, and how does disabling `onCreate` solve it for you?

Hi @manueliglesias 

I'm having the same issue. 

The problem i'm trying to solve is the hard limit of 100 suscriptions for an AppSync API. 

The browser console keeps printing warning logs about disabled suscriptions on my model.

`[WARN] 50:17.608 DataStore - Sync error Connection failed: {"errors":[{"message":"Validation error of type FieldUndefined: Field 'onDeleteKitMaterials' in type 'Subscription' is undefined @ 'onDeleteKitMaterials'"}]}`

**Comment by vrebo** [comment]
I found that in the models/schema.js file the models which should have disabled the subscriptions look like this

```
export const schema = {
  models: {
    Company: {
      name: "Company",
      fields: {...},
      syncable: true, // <------- Here
      pluralName: "Companies",
      attributes: [
        {
          type: "model",
          properties: {
            subscriptions: null,
          },
        },
        ...
      ],
    },
  },
};
```
I changed to **false** the value of the **attribute syncable** in the models that should have disabled the syncronization and now the warnings disappeared. So it works, this is a temporal workaround. 

It would be great if anyone else can confirm this issue too. Here is an example to reproduce it. @manueliglesias 



1. Use the following schema.graphql
```
type Company
  @model(subscriptions: null)
  @auth(
    rules: [
      { allow: private }
    ]
  ) {
  id: ID!
  name: String!
  description: String
  enabled: Boolean
  owner: ID
}
```
2. Generate the DataStore models
`amplify codegen models`

3.- Try any operation on the Company model in the frontend and you'll have an error like this.
`[WARN] 50:17.608 DataStore - Sync error Connection failed: {"errors":[{"message":"Validation error of type FieldUndefined: Field 'onCreateCompany' in type 'Subscription' is undefined @ 'onCreateCompany'"}]}`.

My **amplify-cli version is 4.29.5**.
